libs/server-sent-events/src/client.cpp

```diff
@@ -79,19 +79,45 @@ class FoxyClient : public Client,
           read_timeout_(read_timeout),
           write_timeout_(write_timeout),
           req_(std::move(req)),
-          session_(std::move(executor),
-                   launchdarkly::foxy::session_opts{
-                       ToOptRef(ssl_context_),
-                       connect_timeout.value_or(kNoTimeout)}),
           backoff_(
               initial_reconnect_delay.value_or(kDefaultInitialReconnectDelay),
               kDefaultMaxBackoffDelay),
           last_event_id_(std::nullopt),
-          backoff_timer_(session_.get_executor()),
+          backoff_timer_(std::move(executor)),
           event_receiver_(std::move(receiver)),
           logger_(std::move(logger)),
-          errors_(std::move(errors)) {
+          errors_(std::move(errors)),
+          last_read_(std::nullopt),
+          shutting_down_(false),
+          body_parser_(std::nullopt),
+          session_(std::nullopt) {
         create_parser();
+        create_session();
+    }
+
+    /** Logs a message indicating that an async_read_some operation
+     * on the response's body has completed, and how long that operation took
+     * (using a monotonic clock.)
+     *
+     * This is useful for debugging timeout-related issues, like receiving a
+     * heartbeat message.
+     *
+     * The timestamp of the last read is initialized after receiving successful
+     * headers, but before initiating the first body read. Future reads then
+     * update the timestamp.
+     */
```

Comment:
Added this helper so we can more easily tell how fast / when data is being received. For example, this clearly shows heartbeats arriving every 180 seconds.